packages/expo-splash-screen/CHANGELOG.md

```diff
@@ -7,6 +7,7 @@
 ### 🎉 New features
 
 - Add warning for splash screen if visible for too long ([#12882](https://github.com/expo/expo/pull/12882) by [@ajsmth](https://github.com/ajsmth))
+- Show splash screen without additional setup in MainActivity and could customize resizeMode/statusBarTranslucent in resource. ([#14061](https://github.com/expo/expo/pull/14061) by [@kudo](https://github.com/kudo))
```

Comment:
Isn't it a breaking change 🤔 
I believe it is, because bumping the dependency in the existing project would result in calling `SplashScreen.show(...)` twice (first time from the old Activity-based approach and secondly from the new ActivityLifecycleListener) and I guess that's not something we'd like to deal with? 😓
Moreover I'd like to have some explanation here in CHANGELOG (or meybe not in CHANGELOG (cc @tsapeta)) on how to adopt to the new approach 📖

Reply:
backward compatibility is handled in https://github.com/expo/expo/blob/2fee9a8636cbf70206dfab9278241551eb9c1b93/packages/expo-splash-screen/android/src/main/java/expo/modules/splashscreen/SplashScreenReactActivityLifecycleListener.kt#L13-L17

in expo-splash-screen, calling `show` the second time in showing state is a no-op.
let me know how do you think for this approach.

Reply:
@bbarthec 

> Moreover I'd like to have some explanation here in CHANGELOG (or maybe not in CHANGELOG)

Changelog is not the right place for longer explanations — changelog entries must be concise and generally must fit in one line. If you want to provide more informations, just add something like "More details in the PR" and put more details in its description. Or add a link to expo.fyi document 😉 (more details [here]( https://github.com/expo/expo/blob/master/guides/contributing/Updating%20Changelogs.md))

Reply:
@Kudo that makes sense to me 👍
If it's not breaking anything let's have it that way.
Nevertheless I believe we should write some migration/adoption guide here in PR description (as @tsapeta suggested; I think that expo.fyi is more about scenarios that might happen regularly and not about single change adaption).

Reply:
> @Kudo that makes sense to me 👍
> If it's not breaking anything let's have it that way.
> Nevertheless I believe we should write some migration/adoption guide here in PR description (as @tsapeta suggested; I think that expo.fyi is more about scenarios that might happen regularly and not about single change adaption).

the change is based on the new [`ReactActivityLifecycleListener`](https://github.com/expo/expo/blob/master/packages/expo-modules-core/android/src/main/java/expo/modules/core/interfaces/ReactActivityLifecycleListener.kt) where placing in expo-modules-core. if we need to have the migration guide here, perhaps we also need to talk about unimodules to expo-modules-core migration as well.  how do you think if we have an assumption of the expo-modules-core migration guide will be expo.fyi/somewhere? drafted as following:


# Migration Guide for expo-splash-screen < 0.12.0

The changes in the pr are backward compatible, the code for `expo-splash-screen` will still work as it used to be. However, if you are going to migrate as new style API, here are the steps:

1. [Migrate your project from react-native-unimodules to expo-modules-core](https://expo.fyi/expo-modules-core-migration)
2. Remove expo-splash-screen code from MainActivity

```diff
--- a/android/app/src/main/java/com/helloworld/MainActivity.java
+++ b/android/app/src/main/java/com/helloworld/MainActivity.java
@@ -7,9 +7,6 @@ import com.facebook.react.ReactActivityDelegate;
 import com.facebook.react.ReactRootView;
 import com.swmansion.gesturehandler.react.RNGestureHandlerEnabledRootView;

-import expo.modules.splashscreen.singletons.SplashScreen;
-import expo.modules.splashscreen.SplashScreenImageResizeMode;
-
 public class MainActivity extends ReactActivity {
   @Override
   protected void onCreate(Bundle savedInstanceState) {
     // This is required for expo-splash-screen.
     setTheme(R.style.AppTheme);
     super.onCreate(null);
-    // SplashScreen.show(...) has to be called after super.onCreate(...)
-    // Below line is handled by '@expo/configure-splash-screen' command and it's discouraged to modify it manually
-    SplashScreen.show(this, SplashScreenImageResizeMode.CONTAIN, ReactRootView.class, false);
   }

3. Override default `resizeMode` and `statusBarTranslucent` in stings.xml

```diff
--- a/android/app/src/main/res/values/strings.xml
+++ b/android/app/src/main/res/values/strings.xml
 <?xml version="1.0" encoding="UTF-8" standalone="yes"?>
 <resources>
   <string name="app_name">sdk42</string>
+  <string name="expo_splash_screen_resize_mode">contain</string>
+  <string name="expo_splash_screen_status_bar_translucent">false</string>
</resources>
```

Reply:
@tsapeta have some plans regarding `unimodules`/`expo-modules` migration guide, but I'm not sure where it'll be placed 🤷

Reply:
Regarding the guide you've provided - it looks fine to me 👍 Let's have it in the PR description and we're ready to merge 🤓

Reply:
Oh, and what about the [README installation instruction (both automatic & manual)](https://github.com/expo/expo/tree/master/packages/expo-splash-screen#-configure-android)?

Reply:
> Oh, and what about the [README installation instruction (both automatic & manual)](https://github.com/expo/expo/tree/master/packages/expo-splash-screen#-configure-android)?

that is a great catch 💯  i accidentally overrode and removed my README updates during rebase. this is added now, please take a look. the README seems to be a good place for migration guide. i also add the guide [there](https://github.com/expo/expo/blob/%40kudo/modularize-expo-splash-screen/packages/expo-splash-screen/README.md#%EF%B8%8F-migrate-from-old-versions)

Reply:
Excellent idea 👍